maps/RandomZLevels/hive.dm

```diff
@@ -0,0 +1,482 @@
+			HA.visible_message("<span class='danger'>\The [HA] crawls down from an opening in the ceiling!</span>")
+
+/obj/effect/landmark/hive/monster_spawner
+	name = "alien spawner"
+	desc = "Periodically spawns monsters if the hive replicator isn't destroyed."
+	icon_state = "x"
+
+//Communication unit
+/obj/structure/hive/communicator
+	name = "hive mind"
+	desc = "This must be the main relay that allows different parts of the Hive to communicate with each other through alien pylons."
+	icon_state = "hive_comms"
+	health = 150
+	gibtype = /obj/effect/gibspawner/human
+
+//Husked mob
+/obj/structure/hive/husk
+	name = "hollow husk"
+	desc = "What was once a living creature, now fully dehydrated and mummified by alien technology. Looks very real and alive from a distance."
+	gibtype = /obj/effect/decal/cleanable/scattered_sand
+	health = 1
+
+	//In subtypes, either set the icon and icon state, or set this variable to the mob's path (e.g. /mob/living/simple_animal/hostile/creature)
+	var/mob_type = null
+
+/obj/structure/hive/husk/New()
+	..()
+
+	if(mob_type)
+		var/atom/A = mob_type
+
+		appearance = initial(A.appearance)
+
+		//Because changing the appearance also changes description and name
+		name = initial(name)
+		desc = initial(desc)
+
+	dir = pick(cardinal)
+
+/obj/structure/hive/husk/Die()
+	visible_message("<span class='notice'>\The [src] crumbles into dust!</span>")
+
+	return ..()
+
+/obj/structure/hive/husk/martian //never
+	icon = 'icons/mob/martian.dmi'
+	icon_state = "fuggle"
+
+/obj/structure/hive/husk/martian/New()
+	..()
+
+	icon_state = pick("fuggle", "martian")
+
+/obj/structure/hive/husk/vox
+	icon = 'icons/mob/vox.dmi'
+	icon_state = "armalis"
+
+/obj/structure/hive/husk/creature
+	icon = 'icons/mob/critter.dmi'
+	icon_state = "otherthing_static"
+
+/obj/structure/hive/husk/wolf
+	mob_type = /mob/living/simple_animal/hostile/wolf
+
+//////TRAPS/////////
+/obj/effect/trap/fire_trap //When triggered, spawns a fire blast nearby
+	name = "fire trap"
+	var/fire_type = /obj/effect/fire_blast
+
+	var/fire_damage = 5
+	var/pressure = ONE_ATMOSPHERE * 4.5
+	var/temperature = T0C + 175
+	var/fire_duration
+
+/obj/effect/trap/fire_trap/activate(atom/movable/AM)
+	new /obj/effect/fire_blast/blue(get_step(get_turf(AM), pick(cardinal)), fire_damage, 0, 1, pressure, temperature, fire_duration)
+
+/obj/effect/trap/frog_trap/hive //When triggered, surrounds you with hive denizens
+	name = "monster trap"
+	frog_type = /mob/living/simple_animal/hostile/hive_alien
+
+/obj/effect/trap/frog_trap/hive/artificer //same but artificers
+	frog_type = /mob/living/simple_animal/hostile/hive_alien/artificer
+
+/obj/effect/trap/frog_trap/hive/executioner //same but executioners. VERY dangerous
+	frog_type = /mob/living/simple_animal/hostile/hive_alien/executioner
+
+///////////////////////////////////////////****ITEMS****//////////////////////////////////////////////////
+
+/obj/item/hive/cpu
+	name = "alien processing unit"
```

Comment:
I meant, like what kind of research. Since it's only available on this away mission, it should be tied to a few one of a kind toys, or even an entirely new tech branch for alien technology.